core/src/main/java/org/jruby/RubyTime.java

```diff
@@ -951,7 +951,16 @@ public String toString() {
     @JRubyMethod
     @Override
     public RubyArray to_a(ThreadContext context) {
-        return RubyArray.newArrayNoCopy(context.runtime, sec(), min(), hour(), mday(), month(), year(), wday(), yday(), isdst(), zone());
+        return RubyArray.newArrayNoCopy(
+            context.runtime,
+            sec(), min(), hour(),
+            mday(), month(), year(),
+            wday(), yday(), isdst(),
+            getTimezoneShortName());
+    }
+
+    private RubyString getTimezoneShortName() {
```

Comment:
You should pass runtime in from `to_a` above to avoid retrieving it a second time.